File: 240511/원형 순열에서의 인원 제거/josephus-permutations.java
import java.util.*;

public class Main {
    static Scanner sc = new Scanner(System.in);
    static int n = sc.nextInt();
    static int k = sc.nextInt();
    static Queue<Integer> q = new LinkedList<Integer>();
    static ArrayList<Integer> result = new ArrayList<Integer>();

    public static void main(String[] args) {
        for(int i = 1; i <= n; i++){
            q.add(i);
        }

        while(!q.isEmpty()) {
            for(int i = 1; i < k; i++) {
                q.add(q.poll());
            }
            result.add(q.poll());
        }

        for(int item: result) {
            System.out.print(item + " ");
        }

    }
}

/*
1 2 3 4 5 6
2 3 4 5 6 1
3 4 5 6 1 2
4 5 6 1 2 3
*/

File: 240511/원형 순열에서의 인원 제거/josephus-permutations.kt
import java.util.*

val sc = Scanner(System.`in`)
val n = sc.nextInt()
val k = sc.nextInt()
val queue: Queue<Int> = LinkedList<Int>()
val result = mutableListOf<Int>()

fun main() {
    for(i in 1 .. n) {
        queue.add(i)
    }

    while(!queue.isEmpty()) {
        for(i in 1 until k) {
            queue.add(queue.poll())
        }
        result.add(queue.poll())
    }

    for(item in result){
        print("$item ")
    }
}